src/logger.cc

```diff
@@ -50,6 +50,41 @@ NAN_METHOD(setLevel) {
   spdlog::set_level(level);
 }
 
+NAN_METHOD(setFlushOn) {
+  if (!info[0]->IsNumber()) {
+    return Nan::ThrowError(Nan::Error("Provide flush level"));
+  }
+
+  const int64_t levelNumber = Nan::To<int64_t>(info[0]).FromJust();
+  spdlog::level::level_enum level;
+  switch (levelNumber) {
```

Comment:
I would move this switch block into a separate function, but I'm not sure how to handle it returning both a `spdlog::level::level_enum` or a Nan Error.

Reply:
The switch block can be replaced with

```c++
auto level = static_cast<spdlog::level::level_enum>(levelNumber);
if (level >= spdlog::level::n_levels) {
  return Nan::ThrowError(Nan::Error("Invalid level"));
}
```

also change it in other places where this switch block is repeated.

Reply:
I changed it around a bit. It seems that directly casting to an enum like that results in version-dependent behaviour: https://stackoverflow.com/questions/18195312/what-happens-if-you-static-cast-invalid-value-to-enum-class.